akka-stream/src/main/scala/akka/stream/remote/serialization/StreamRefSerializer.scala

```diff
@@ -0,0 +1,183 @@
+    case ref: SourceRef[_]                 ⇒ serializeSourceRef(ref).toByteArray
+  }
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = manifest match {
+    // protocol
+    case SequencedOnNextManifest     ⇒ deserializeSequencedOnNext(bytes)
+    case CumulativeDemandManifest    ⇒ deserializeCumulativeDemand(bytes)
+    case RemoteSinkCompletedManifest ⇒ deserializeRemoteSinkCompleted(bytes)
+    case RemoteSinkFailureManifest   ⇒ deserializeRemoteSinkFailure(bytes)
+    // refs
+    case SinkRefManifest             ⇒ deserializeSinkRef(bytes)
+    case SourceRefManifest           ⇒ ???
+  }
+
+  // -----
+
+  private def serializeCumulativeDemand(d: StreamRefs.CumulativeDemand): StreamRefContainers.CumulativeDemand = {
+    StreamRefContainers.CumulativeDemand.newBuilder()
+      .setSeqNr(d.seqNr)
+      .build()
+  }
+
+  private def serializeRemoteSinkFailure(d: StreamRefs.RemoteSinkFailure): StreamRefContainers.RemoteSinkFailure = {
+    StreamRefContainers.RemoteSinkFailure.newBuilder()
+      .setCause(ByteString.copyFrom(d.msg.getBytes))
+      .build()
+  }
+
+  private def serializeRemoteSinkCompleted(d: StreamRefs.RemoteSinkCompleted): StreamRefContainers.RemoteSinkCompleted = {
+    StreamRefContainers.RemoteSinkCompleted.newBuilder()
+      .setSeqNr(d.seqNr)
+      .build()
+  }
+
+  private def serializeSequencedOnNext(o: StreamRefs.SequencedOnNext[_]) = {
+    val p = o.payload.asInstanceOf[AnyRef]
+    val msgSerializer = serialization.findSerializerFor(p)
+
+    val payloadBuilder = StreamRefContainers.Payload.newBuilder()
+      .setEnclosedMessage(ByteString.copyFrom(msgSerializer.toBinary(p)))
+      .setSerializerId(msgSerializer.identifier)
+
+    msgSerializer match {
+      case ser2: SerializerWithStringManifest ⇒
+        val manifest = ser2.manifest(p)
+        if (manifest != "")
+          payloadBuilder.setMessageManifest(ByteString.copyFromUtf8(manifest))
+      case _ ⇒
+        if (msgSerializer.includeManifest)
+          payloadBuilder.setMessageManifest(ByteString.copyFromUtf8(p.getClass.getName))
+    }
+
+    StreamRefContainers.SequencedOnNext.newBuilder()
+      .setSeqNr(o.seqNr)
+      .setPayload(payloadBuilder.build())
+      .build()
+  }
+
+  private def serializeSinkRef(sink: SinkRef[_]): StreamRefContainers.SinkRef = {
+    val make = () ⇒ {
+      val actorRef = StreamRefContainers.ActorRef.newBuilder()
+        .setPath(Serialization.serializedActorPath(sink.targetRef))
+
+      StreamRefContainers.SinkRef.newBuilder()
+        .setInitialDemand(sink.initialDemand)
+        .setTargetRef(actorRef)
+        .build()
+    }
+
+    // Serialize actor references with full address information (defaultAddress).
+    // When sending remote messages currentTransportInformation is already set,
+    // but when serializing for digests it must be set here.
+    val it = if (Serialization.currentTransportInformation.value == null)
```

Comment:
I don't think this stuff is needed. This serializer is only used in remoting and then the transportInfo is always set. This looks like a copy from ddata, where that serializer is also used outside remoting.

Reply:
Yeah, took it from ddata indeed since was having issues with serialization, which we tracked down with Johannes yesterday to be a log statement that consumed an iterator so all the actor path resolving broke 😱 I though maybe the data inserted was wrong at some point, so that's why this. Removing then!